modules/completion/init.zsh

```diff
@@ -15,7 +15,15 @@ fi
 fpath=("${0:h}/external/src" $fpath)
 
 # Load and initialize the completion system ignoring insecure directories.
-autoload -Uz compinit && compinit -i
+# Will only rebuild ~/.zcompdump once per day
+autoload -Uz compinit
+if [ $(date +'%j') != $(stat -f '%Sm' -t '%j' ~/.zcompdump) ]; then
```

Comment:
I also like the spirit of this pull and was running it my personal fork at some point.  I run all BSDs so didn't notice, but if there's a reliable command that errors out on BSD and works on GNU couldn't one simply `stat --version 2>/dev/null && foo=yes` or `stats --version 2>/dev/null; if $? = 0...`